pkg/controller/identity/iamrolepolicyattachment/controller.go

```diff
@@ -91,13 +91,13 @@ type external struct {
 }
 
 func (e *external) Observe(ctx context.Context, mgd resource.Managed) (managed.ExternalObservation, error) {
-	cr, ok := mgd.(*v1alpha3.IAMRolePolicyAttachment)
+	cr, ok := mgd.(*v1beta1.IAMRolePolicyAttachment)
 	if !ok {
 		return managed.ExternalObservation{}, errors.New(errUnexpectedObject)
 	}
 
 	req := e.client.ListAttachedRolePoliciesRequest(&awsiam.ListAttachedRolePoliciesInput{
-		RoleName: aws.String(cr.Spec.RoleName),
+		RoleName: aws.String(cr.Spec.ForProvider.RoleName),
 	})
 
 	observed, err := req.Send(ctx)
```

Comment:
err check here can also be refactored as described in comment on iam role controller